contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2880,16 +2880,181 @@ add_entry_to_bbf_schema(const char *schema_name,
 	CommandCounterIncrement();
 }
 
+/*
+ * Get the value of permission column from BABELFISH_SCHEMA_PERMISSIONS table.
+ */
+int16
+get_bbf_schema_privilege(const char *schema_name,
+					const char *object_name,
+					const char *grantee)
+{
+	Relation	bbf_schema_rel;
+	HeapTuple	tuple_bbf_schema;
+	ScanKeyData	scanKey[4];
+	SysScanDesc	scan;
+	int16	dbid = get_cur_db_id();
+	int16 permission = 0;
+
+	bbf_schema_rel = table_open(get_bbf_schema_perms_oid(),
+									AccessShareLock);
+	ScanKeyInit(&scanKey[0],
+				Anum_bbf_schema_perms_dbid,
+				BTEqualStrategyNumber, F_INT2EQ,
+				Int16GetDatum(dbid));
+	ScanKeyEntryInitialize(&scanKey[1], 0,
+				Anum_bbf_schema_perms_schema_name,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
+				CStringGetDatum(schema_name));
+	ScanKeyEntryInitialize(&scanKey[2], 0,
+				Anum_bbf_schema_perms_object_name,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
+				CStringGetDatum(object_name));
+	ScanKeyEntryInitialize(&scanKey[3], 0,
+				Anum_bbf_schema_perms_grantee,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
+				CStringGetDatum(grantee));
```

Comment:
Based on this logic you are going to return the first object which matches irrespective of the object_type.
What if the first one was a table but user was granting for a procedure?